include/small/abstract_layer.hpp

```diff
@@ -1201,5 +1837,4519 @@ void abstract_layer(
+                        AccumT *O_col_left = O_col_full + O_w_full * (_G_b * _K_b); // ScalarT --> AccumT
+
+#if DEBUG
+                        printf(" calling right\n");
+#endif
+                        kernel_right_1D<ScalarT, AccumT,
+                                        _G_b, _K_b, _F_cb, _O_wb, _stride,
+                                        _UNROLL, op_type, op_class,
+                                        op_fused_single_element_before,
+                                        op_fused_single_element_after,
+                                        _stride_before, _stride_after>(
+                                            first,
+                                            //F_h,
+                                            F_w,
+                                            I_w * _C_ib,
+                                            O_w_left,
+                                            r_pad_el,
+                                            pad_right,
+                                            I_col_left,
+                                            F_col_left,
+                                            O_col_left,
+                                            //0,
+                                            //0,
+                                            F_before_buf_group,
+                                            F_after_buf_group);
+                    }
+                }
+            }
+        }
+    }
+}
+
+//****************************************************************************
+// Overloading to fuse partial spatial, single channel as well as partial spatial partial channel reductions
+//****************************************************************************
+//****************************************************************************
+// Fused abstract layer
+//****************************************************************************
+template <typename BufferT,
+          dim_t _G_b,
+          dim_t _K_b,
+          dim_t _F_cb,
+          dim_t _O_wb,
+          dim_t _stride,
+          dim_t _UNROLL,
+          OpType op_type,
+          int8_t op_class, //  2  (conv),  1  (dense,pool), or '0' (activation, upsample)
+          bool rewrite_output,
+
+          dim_t _G_b_1,
+          dim_t _K_b_1,
+          dim_t _F_cb_1,
+          dim_t _O_wb_1,
+          dim_t _stride_1,
+          dim_t _UNROLL_1,
+          OpType op_type_1,
+          int8_t op_class_1, //  2  (conv),  1  (dense,pool), or '0' (activation, upsample)
+          bool rewrite_output_1,
+
+          OpType op_fused_single_element_before = OP_NONE,
+          OpType op_fused_single_element_after = OP_NONE,
+          dim_t _stride_before = 1,
+          dim_t _stride_after = 1,
+
+          OpType op_fused_single_element_before_1 = OP_NONE,
+          OpType op_fused_single_element_after_1 = OP_NONE,
+          dim_t _stride_before_1 = 1,
+          dim_t _stride_after_1 = 1>
+void fused_abstract_layer_1D(
```

Comment:
Why do we need two functions. Are they both used?  Delete one if not.  This one is suspect as it calls kernel functions that are not 1D variants.